### PR TITLE
feat(command): Implement GET /device/all V2 API

### LINKED ISSUE #3152: [V2 command] Implement GET /device/all
# 🚀 Feature Request

### Relevant Package

 This feature request is for core command.

### Description
 
Per https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-command/2.x#/default/get_device_all, implement GET /device/all V2 API.

### PR DESCRIPTION
**Pull request by judehung**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/master/.github/Contributing.md.

## Issue Number:
fix #3152 

## What is the new behavior?
Per https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-command/2.x#/default/get_device_all, implement GET /device/all V2 API.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by judehung** [comment]
This PR depends on https://github.com/edgexfoundry/go-mod-core-contracts/pull/507, and go.mod shall be updated after dependency is merged.

**Comment by judehung** [comment]
rebased and updated go.mod

File: openapi/v2/core-command.yaml [hunk 1]
```diff
@@ -44,13 +44,25 @@ components:
...
+          type: string
+        profileName:
+          type: string
+        coreCommands:
+          type: array
+          items:
+            $ref: '#/components/schemas/CoreCommand'
+      required:
+        - deviceName
+        - profileName
```
Comment (lenny-intel) [inline openapi/v2/core-command.yaml:60]: Don't need the required section on Response only objects. i.e no validate performed as we build the response DTO.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: openapi/v2/core-command.yaml (hunk 1 above)
Comment (judehung) [reply openapi/v2/core-command.yaml]: Agree.  I've removed required section out of response DTO.  Please check https://github.com/edgexfoundry/edgex-go/pull/3153/commits/47b4c9e6fc4044b20e859d8e973f792df8961e2f

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**